src/panels/SkyBrowserPanel/hooks.ts

```diff
@@ -0,0 +1,223 @@
+    return () => {
+      dispatch(unsubscribeToSkyBrowser());
+    };
+  }, [dispatch]);
+}
+
+export function useActiveImage(): [string, (url: string) => void] {
+  const activeImage = useAppSelector((state) => state.skybrowser.activeImage);
+
+  const dispatch = useAppDispatch();
+
+  function setImage(url: string) {
+    dispatch(setActiveImage(url));
+  }
+  return [activeImage, setImage];
+}
+
+// Below are hooks to get the values from the sky browser topic. They
+// are setup so the value from the selector only should update when
+// the value itself actually updates. TODO (ylvse 2025-01-29): rewrite
+// the topic so that most of the data are properties instead, and only
+// pass the data for the currently selected browser
+
+export function useSelectedBrowserColor(): number[] | undefined {
+  const color = useAppSelector(
+    (state) => state.skybrowser.selectedBrowser?.color,
+    lowPrecisionEqualArray
+  );
+  return color;
+}
+
+export function useSelectedBrowserColorString(): string | undefined {
+  const color = useSelectedBrowserColor();
+  return color ? (`rgb(${color.join(',')})` as string) : undefined;
+}
+
+export function useSelectedBrowserRadius(): number | undefined {
+  const radius = useAppSelector(
+    (state) => state.skybrowser.selectedBrowser?.borderRadius,
+    lowPrecisionEqual()
+  );
+  return radius;
+}
+
+export function lowPrecisionEqualMatrix(
+  lhs: number[][] | undefined,
+  rhs: number[][] | undefined
+) {
+  if (lhs === undefined && rhs === undefined) {
+    return true;
+  }
+  if (lhs === undefined || rhs === undefined) {
+    return false;
+  }
+  if (lhs.length !== rhs.length) {
+    return false;
+  }
+  return lhs.every((value, i) => lowPrecisionEqualArray(value, rhs[i]));
+}
+
+export function equalArray<T>(lhs: T[] | undefined, rhs: T[] | undefined) {
+  if (lhs === undefined && rhs === undefined) {
+    return true;
+  }
+  if (lhs === undefined || rhs === undefined) {
+    return false;
+  }
+  if (lhs.length !== rhs.length) {
+    return false;
+  }
+  return lhs.every((value, i) => value === rhs[i]);
+}
+
+export function lowPrecisionEqualArray(
+  lhs: number[] | undefined,
+  rhs: number[] | undefined
+) {
+  if (lhs === undefined && rhs === undefined) {
+    return true;
+  }
+  if (lhs === undefined || rhs === undefined) {
+    return false;
+  }
+  if (lhs.length !== rhs.length) {
+    return false;
+  }
+
+  return lhs?.every((value, i) => lowPrecisionEqual()(value, rhs[i]));
+}
+
+export function lowPrecisionEqual(epsilon: number = 1e-3) {
+  return (lhs: number | undefined, rhs: number | undefined) => {
+    if (lhs === undefined && rhs === undefined) {
+      return true;
+    }
+    if (lhs === undefined || rhs === undefined) {
+      return false;
+    }
+    return Math.abs(lhs - rhs) < epsilon;
+  };
+}
+
+export function useSelectedBrowserCoords() {
+  const ra = useAppSelector(
+    (state) => state.skybrowser.selectedBrowser?.ra,
+    lowPrecisionEqual(1e-6)
```

Comment:
Why is this specific precision needed for ra dec and roll?  Would be a lot nicer if the `lowPrecisionEqual` function didn't have to work differently than the other equality functions